src/java.base/share/classes/java/io/ObjectStreamClass.java

```diff
@@ -459,6 +461,10 @@ public Void run() {
             }
         }
         initialized = true;
+
+        if (SerializationMisdeclarationEvent.enabled() && serializable) {
+            new SerializationMisdeclarationChecker(cl).checkMisdeclarations();
```

Comment:
Is there any benefit to avoiding the allocation and passing the class through the methods as an extra arg?

Reply:
The allocation has been removed and the event messages have been simplified in the most recent commit.